Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/DriverOnboarding/Status.hs

```diff
@@ -258,18 +259,24 @@ activateRCAutomatically personId merchantId merchantOpCityId rcNumber = do
           }
   void $ DomainRC.linkRCStatus (personId, merchantId, merchantOpCityId) rcStatusReq
 
-checkIfDocumentValid :: Id DMOC.MerchantOperatingCity -> DVC.DocumentType -> DVeh.Category -> ResponseStatus -> Flow Bool
-checkIfDocumentValid merchantOperatingCityId docType category status = do
+checkIfDocumentValid :: Id DMOC.MerchantOperatingCity -> DVC.DocumentType -> DVeh.Category -> ResponseStatus -> Id SP.Person -> Flow Bool
+checkIfDocumentValid merchantOperatingCityId docType category status personId = do
   mbVerificationConfig <- CQDVC.findByMerchantOpCityIdAndDocumentTypeAndCategory merchantOperatingCityId docType category
   case mbVerificationConfig of
     Just verificationConfig -> do
       if verificationConfig.isMandatory
         then case status of
-          VALID -> return True
+          VALID -> do
+            isSubscriptionValid <- if docType == DVC.SubscriptionPlan then validateSubscriptionStatus else pure True
```

Comment:
This should be handled here, not in this line
https://github.com/nammayatri/nammayatri/pull/7045/files#diff-bc0e75cbb82db8b623e91a8a31d398d510d48adb9408c35e452e9a2e8a6b5468R297